Commands/Files/AddFile.cs

```diff
@@ -124,6 +132,47 @@ protected override void ExecuteCmdlet()
 
                 ClientContext.ExecuteQueryRetry();
             }
+            if (ContentType != null)
+            {
+                ClientContext.Load(file, f=>f.ListId);
+                ClientContext.ExecuteQueryRetry();
+
+                List list = SelectedWeb.Lists.GetById(file.ListId);
+                ClientContext.Load(list);
+                ClientContext.ExecuteQueryRetry();
+
+                ContentType ct;
+                if (!string.IsNullOrEmpty(ContentType.Id))
+                {
+                    ct = list.GetContentTypeById(ContentType.Id);
+                }
+                else if(!string.IsNullOrEmpty(ContentType.Name))
+                {
+                    ct = list.GetContentTypeByName(ContentType.Name);
+                }
+                else if(ContentType.ContentType != null)
+                {
+                    ct = ContentType.ContentType;
+                }
+                else
+                {
+                    ct = null;
+                   
+                }
+
+                if (ct != null)
+                {
+                    var item = file.ListItemAllFields;
+                    item["ContentTypeId"] = ct.Id.StringValue;
+                    item.Update();
+                    ClientContext.ExecuteQueryRetry();
+                }
+                else
+                {
+                    WriteVerbose("Content Type was 'null' was not found.");
```

Comment:
As such throwing an error is a good solution, however consider going for WriteError or ThrowTerminatingError. See these links: 
https://blogs.msdn.microsoft.com/kebab/2013/06/09/an-introduction-to-error-handling-in-powershell/
https://msdn.microsoft.com/en-us/library/ms714414(v=vs.85).aspx